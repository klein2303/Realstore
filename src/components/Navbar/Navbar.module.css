.navbar {
    background-color: white;
    width: 100%;
    height: 120px;
    position: relative;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;
    padding-top: 20px;
    font-family: "Source Sans 3", sans-serif;
}

.navpage {
    height: max-content;
    width: 40%;
    padding-left: 20px;

    display: flex;
    flex-direction: row;
    justify-content: start;
    align-items: center;
    gap: 8px;
}

.navlogo {
    padding: 10px;
    width: 20%;
    height: max-content;

    display: flex;
    justify-content: center;
    align-items: center;
}

.logoimg {
    width: 90px;
    height: auto;
}

.navlinks {
    display: flex;
    flex-direction: row-reverse;
    justify-content: end;
    align-items: center;
    width: 40%;
    height: 90px;
}

.navlink,
.navlink:hover,
.navlink:focus {
    text-decoration: none;
    color: #424242;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
    margin: 10px;
    font-size: 18px;
}

.navlink:hover {
    text-decoration: underline;
}

.cartbutton,
.cartbutton:hover,
.cartbutton:focus {
    text-decoration: none;
    color: white;
    background-color: #424242;
    height: 48px;
    display: flex;
    flex-direction: row;
    align-items: center;
    margin: 10px;
    padding-left: 25px;
    padding-right: 25px;
    font-size: 14px;
    gap: 10px;
    border-radius: 30px;
}

.cartbutton:hover {
    background-color: #616161;
}

.heart {
    font-size: 36px;
    color: #424242;
}

.heart:hover {
    color: red;
}

.ulstyle {
    list-style-type: none;
}

.whiteicon {
    font-size: 1.6em;
    color: white;
}

.hamburgmenu {
    display: none;
}

.shoppingButton {
    display: none;
}

@media (max-width: 800px) {
    .navpage {
        display: none;
    }

    .hamburgmenu {
        display: flex;
        justify-content: start;
        align-items: center;
        font-size: 1.6em;
        color: #424242;
        width: 65px;
        cursor: pointer;
    }

    .navlinks {
        display: none;
    }

    .crosspos {
        display: flex;
        justify-content: start;
        width: 100%;
        padding-top: 22px;
        padding-bottom: 20px;
    }

    .cross {
        font-size: 1.6em;
        color: #424242;
        cursor: pointer;
    }

    .shoppingButton {
        text-decoration: none;
        color: white;
        background-color: #424242;
        height: 30px;

        display: flex;
        flex-direction: row;
        align-items: center;
        justify-content: space-between;
        margin: 10px;
        padding-left: 12px;
        padding-right: 12px;
        font-size: 14px;

        border-radius: 30px;
    }

    .navlink {
        width: 100%;
        height: 20px;
    }

    .navbar {
        background-color: white;
        width: 100%;
        height: 70px;
        position: relative;
        gap: 20%;
        display: flex;
        flex-direction: row;
        align-items: center;
        justify-content: center;
    }

    .logoimg {
        width: 60px;
        height: auto;
    }

    .mobilemenu {
        background-color: white;
        display: flex;
        flex-direction: column;
        justify-content: start;
        text-align: start;
        align-items: center;
        position: absolute;
        left: 0;
        top: 20px;
        margin-left: 20px;
        height: fit-content;
        width: 125px;
        z-index: 50;
    }
}
