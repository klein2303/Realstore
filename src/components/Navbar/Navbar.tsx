import {CiHeart} from "react-icons/ci";
import {HiOutlineShoppingBag} from "react-icons/hi2";
import {Link} from "react-router-dom";
import styles from "./Navbar.module.css";
import {RxHamburgerMenu, RxCross1} from "react-icons/rx";
import {useState} from "react";
import logo from "/Logo.svg";

const Navbar = () => {
    //For Hamburgermenu in mobile-version
    const [isOpen, setIsOpen] = useState(false);

    return (
        <main>
            <nav className={styles.navbar} role="navigation" aria-label="Main Navigation">
                {isOpen && (
                    <div className={styles.mobilemenu} role="dialog" aria-label="Mobile Navigation Menu">
                        <div className={styles.crosspos}>
                            <RxCross1
                                className={styles.cross}
                                onClick={() => setIsOpen(false)}
                                aria-label="Close Menu"
                            />
                        </div>
                        <Link to={"/"} className={styles.navlink}>
                            <p>HOME</p>
                        </Link>
                        <Link to={"/Shopping"} className={styles.navlink}>
                            <p>SHOP</p>
                        </Link>
                        <Link to={"/Favorites"} className={styles.navlink}>
                            <p>FAVORITES</p>
                        </Link>
                    </div>
                )}
                <RxHamburgerMenu
                    className={styles.hamburgmenu}
                    onClick={() => setIsOpen(true)}
                    aria-label="Open Menu"
                />
                <div className={styles.navpage} role="navigation" aria-label="Desktop Navigation Links">
                    <Link to={"/"} className={styles.navlink}>
                        <p>HOME</p>
                    </Link>
                    <Link to={"/Shopping"} className={styles.navlink}>
                        <p>SHOP</p>
                    </Link>
                </div>
                <div className={styles.navlogo}>
                    <Link to={"/"}>
                        <img src={logo} alt="RealStore logo" className={styles.logoimg} aria-label="RealStore Logo" />
                    </Link>
                </div>
                <div className={styles.navlinks} role="navigation" aria-label="User Navigation">
                    <Link to={"/ShoppingCart"} className={styles.cartbutton}>
                        <p>MY CART</p>
                        <HiOutlineShoppingBag className={styles.whiteicon} />
                    </Link>
                    <Link to={"/Favorites"} className={styles.navlink}>
                        <CiHeart className={styles.heart} aria-label="View My Favorites" />
                    </Link>
                </div>
                <Link to={"/ShoppingCart"} className={styles.shoppingButton} aria-label="View My Cart">
                    <HiOutlineShoppingBag className={styles.whiteicon} />
                </Link>
            </nav>
        </main>
    );
};

export default Navbar;
