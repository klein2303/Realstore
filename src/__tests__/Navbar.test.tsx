import {render, screen} from "@testing-library/react";
import "@testing-library/jest-dom"; // Import jest-dom for extended matchers
import Navbar from "../components/Navbar/Navbar"; // Adjust the import path
import {MemoryRouter} from "react-router-dom";

// Helper function to render the Navbar component
const renderNavbar = () => {
    return render(
        <MemoryRouter>
            <Navbar />
        </MemoryRouter>,
    );
};

test("Navbar snapshot", () => {
    const {asFragment} = renderNavbar();
    expect(asFragment()).toMatchSnapshot();
});

test("renders all navigation links", () => {
    renderNavbar();

    // Check for the presence of navigation links
    expect(screen.getByText(/HOME/i)).toBeInTheDocument();
    expect(screen.getByText(/SHOP/i)).toBeInTheDocument();
    expect(screen.getByText(/MY CART/i)).toBeInTheDocument();
    expect(screen.getByAltText(/RealStore logo/i)).toBeInTheDocument();
});
