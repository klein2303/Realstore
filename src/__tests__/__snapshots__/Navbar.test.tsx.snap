// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar snapshot 1`] = `
<DocumentFragment>
  <main>
    <nav
      aria-label="Main Navigation"
      class="_navbar_278c00"
      role="navigation"
    >
      <svg
        aria-label="Open Menu"
        class="_hamburgmenu_278c00"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 15 15"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M1.5 3C1.22386 3 1 3.22386 1 3.5C1 3.77614 1.22386 4 1.5 4H13.5C13.7761 4 14 3.77614 14 3.5C14 3.22386 13.7761 3 13.5 3H1.5ZM1 7.5C1 7.22386 1.22386 7 1.5 7H13.5C13.7761 7 14 7.22386 14 7.5C14 7.77614 13.7761 8 13.5 8H1.5C1.22386 8 1 7.77614 1 7.5ZM1 11.5C1 11.2239 1.22386 11 1.5 11H13.5C13.7761 11 14 11.2239 14 11.5C14 11.7761 13.7761 12 13.5 12H1.5C1.22386 12 1 11.7761 1 11.5Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
      <div
        aria-label="Desktop Navigation Links"
        class="_navpage_278c00"
        role="navigation"
      >
        <a
          class="_navlink_278c00"
          href="/"
        >
          <p>
            HOME
          </p>
        </a>
        <a
          class="_navlink_278c00"
          href="/Shopping"
        >
          <p>
            SHOP
          </p>
        </a>
      </div>
      <div
        class="_navlogo_278c00"
      >
        <a
          href="/"
        >
          <img
            alt="RealStore logo"
            aria-label="RealStore Logo"
            class="_logoimg_278c00"
            src="/project1/Logo.svg"
          />
        </a>
      </div>
      <div
        aria-label="User Navigation"
        class="_navlinks_278c00"
        role="navigation"
      >
        <a
          class="_cartbutton_278c00"
          href="/ShoppingCart"
        >
          <p>
            MY CART
          </p>
          <svg
            aria-hidden="true"
            class="_whiteicon_278c00"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.75 10.5V6a3.75 3.75 0 1 0-7.5 0v4.5m11.356-1.993 1.263 12c.07.665-.45 1.243-1.119 1.243H4.25a1.125 1.125 0 0 1-1.12-1.243l1.264-12A1.125 1.125 0 0 1 5.513 7.5h12.974c.576 0 1.059.435 1.119 1.007ZM8.625 10.5a.375.375 0 1 1-.75 0 .375.375 0 0 1 .75 0Zm7.5 0a.375.375 0 1 1-.75 0 .375.375 0 0 1 .75 0Z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </a>
        <a
          class="_navlink_278c00"
          href="/Favorites"
        >
          <svg
            aria-label="View My Favorites"
            class="_heart_278c00"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              id="Heart"
            >
              <path
                d="M12,20.043a.977.977,0,0,1-.7-.288L4.63,13.08A5.343,5.343,0,0,1,6.053,4.513,5.266,5.266,0,0,1,12,5.371a5.272,5.272,0,0,1,5.947-.858A5.343,5.343,0,0,1,19.37,13.08l-6.676,6.675A.977.977,0,0,1,12,20.043ZM8.355,4.963A4.015,4.015,0,0,0,6.511,5.4,4.4,4.4,0,0,0,4.122,8.643a4.345,4.345,0,0,0,1.215,3.73l6.675,6.675,6.651-6.675a4.345,4.345,0,0,0,1.215-3.73A4.4,4.4,0,0,0,17.489,5.4a4.338,4.338,0,0,0-4.968.852h0a.744.744,0,0,1-1.042,0A4.474,4.474,0,0,0,8.355,4.963Z"
              />
            </g>
          </svg>
        </a>
      </div>
      <a
        aria-label="View My Cart"
        class="_shoppingButton_278c00"
        href="/ShoppingCart"
      >
        <svg
          aria-hidden="true"
          class="_whiteicon_278c00"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.75 10.5V6a3.75 3.75 0 1 0-7.5 0v4.5m11.356-1.993 1.263 12c.07.665-.45 1.243-1.119 1.243H4.25a1.125 1.125 0 0 1-1.12-1.243l1.264-12A1.125 1.125 0 0 1 5.513 7.5h12.974c.576 0 1.059.435 1.119 1.007ZM8.625 10.5a.375.375 0 1 1-.75 0 .375.375 0 0 1 .75 0Zm7.5 0a.375.375 0 1 1-.75 0 .375.375 0 0 1 .75 0Z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </a>
    </nav>
  </main>
</DocumentFragment>
`;
